python/dgl/graph_index.py

```diff
@@ -900,12 +952,14 @@ def create_graph_index(graph_data=None, multigraph=False, readonly=False):
         return graph_data
 
     if readonly:
-        return create_immutable_graph_index(graph_data)
-
-    handle = _CAPI_DGLGraphCreate(multigraph)
-    gidx = GraphIndex(handle)
+        gidx = GraphIndex(None, multigraph, readonly)
```

Comment:
To avoid this awkward `None`, the solution is to first process all the graph data. There are functions to convert different types of graph data to `src, dst, edge_id, num_nodes` first, so the handler could be created and then the python shim `GraphIndex`. Leave this for now. But would you put a `FIXME` here?